src/arrayMethodSort.js

```diff
@@ -4,8 +4,32 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const defaultCompareFunction = (a, b) => {
+      return a.toString() > b.toString()
+        ? 1
+        : a.toString() < b.toString()
+          ? -1
+          : 0;
+    };
+
+    const comparator =
+      typeof compareFunction === 'function'
+        ? compareFunction
+        : defaultCompareFunction;
+
+    for (let i = 1; i < this.length; i++) {
+      const key = this[i];
+      let j = i - 1;
+
+      while (j >= 0 && comparator(this[j], key) > 0) {
+        this[j + 1] = this[j];
+        j--;
+      }
+      this[j + 1] = key;
+    }
+
+    return this;
```

Comment:
The current implementation uses an insertion sort algorithm. While this is acceptable, consider using a more efficient sorting algorithm like quicksort or mergesort for better performance on larger arrays.